packages/frontend-2/components/viewer/PreSetupWrapper.vue

```diff
@@ -0,0 +1,153 @@
+        </Portal>
+
+        <ClientOnly>
+          <!-- Tour host -->
+          <div
+            v-if="showTour"
+            class="fixed w-full h-[100dvh] flex justify-center items-center pointer-events-none z-[100]"
+          >
+            <TourOnboarding />
+          </div>
+          <!-- Viewer host -->
+          <div
+            class="special-gradient absolute z-10 overflow-hidden w-screen"
+            :class="
+              isEmbedEnabled
+                ? isTransparent
+                  ? 'viewer-transparent h-[100dvh]'
+                  : 'h-[calc(100dvh-3.5rem)]'
+                : 'h-[100dvh]'
+            "
+          >
+            <ViewerBase />
+            <Transition
+              enter-from-class="opacity-0"
+              enter-active-class="transition duration-1000"
+            >
+              <ViewerAnchoredPoints v-show="showControls" />
+            </Transition>
+          </div>
+
+          <!-- Global loading bar -->
+          <ViewerLoadingBar class="z-20" />
+
+          <!-- Sidebar controls -->
+          <Transition
+            enter-from-class="opacity-0"
+            enter-active-class="transition duration-1000"
+          >
+            <ViewerControls v-show="showControls" class="z-20" />
+          </Transition>
+
+          <!-- Viewer Object Selection Info Display -->
+          <Transition
+            v-if="!hideSelectionInfo"
+            enter-from-class="opacity-0"
+            enter-active-class="transition duration-1000"
+          >
+            <div v-show="showControls">
+              <ViewerSelectionSidebar class="z-20" />
+            </div>
+          </Transition>
+          <!-- Shows up when filters are applied for an easy return to normality -->
+          <ViewerGlobalFilterReset class="z-20" :embed="!!isEmbedEnabled" />
+        </ClientOnly>
+      </div>
+    </ViewerPostSetupWrapper>
+    <ViewerEmbedFooter
+      :name="modelName || 'Loading...'"
+      :date="lastUpdate"
+      :url="route.path"
+    />
+    <Portal to="primary-actions">
+      <HeaderNavShare
+        v-if="project"
+        :resource-id-string="modelId"
+        :version-id="versionId"
+        :project-id="project.id"
+        :visibility="project.visibility"
+      />
+    </Portal>
+  </div>
+</template>
+<script setup lang="ts">
+import {
+  useSetupViewer,
+  type InjectableViewerState
+} from '~~/lib/viewer/composables/setup'
+import dayjs from 'dayjs'
+import { graphql } from '~~/lib/common/generated/gql'
+import { useEmbed } from '~/lib/viewer/composables/setup/embed'
+import { useViewerTour } from '~/lib/viewer/composables/tour'
+
+const emit = defineEmits<{
+  setup: [InjectableViewerState]
+}>()
+
+const route = useRoute()
+const { showTour, showControls } = useViewerTour()
+
+const projectId = computed(() => route.params.id as string)
+const modelId = computed(() => route.params.modelId as string)
+const versionId = computed(() => {
+  const parts = modelId.value.split('@')
```

Comment:
1. Use resourceIdString builder utils for this
2. There could easily be multiple versionIds in the url (multiple models). How is this supposed to work?

Reply:
This was an unused prop, so i removed